controller_cat.go

```diff
@@ -138,16 +134,16 @@ func (c *controller) asyncPeerRequest(peer *Peer) ([]Endpoint, error) {
 		c.shareMtx.Unlock()
 	}()
 
-	req := newParcel(TypePeerRequest, []byte("Peer Request"))
+	req := newParcel(TypePeerRequest, nil)
 	peer.Send(req)
 
 	select {
-	case <-async:
+	case parcel := <-async:
+		share := c.trimShare(c.processPeerShare(peer, parcel), true)
```

Comment:
I changed the base of the PR to `develop`, which means you'll have to change `trimShare` to `shuffleTrimShare`. sorry!